mars/dataframe/indexing/iloc.py

```diff
@@ -0,0 +1,244 @@
+            else:
+                shape = tensor1.shape
+                dtype = find_common_type(df.dtypes.iloc[self.indexes[1]].values)
+                index_value = indexing_index_value(df.columns, self.indexes[1])
+            self._object_type = ObjectType.series
+            return self.new_series([df], shape=shape, dtype=dtype, index_value=index_value)
+        else:
+            return self.new_dataframe([df], shape=tensor0.shape + tensor1.shape, dtypes=df.dtypes.iloc[self.indexes[1]],
+                                      index_value=indexing_index_value(df.index_value, self.indexes[0]),
+                                      columns_value=indexing_index_value(df.columns, self.indexes[1], store_data=True))
+
+    def on_output_modify(self, new_output):
+        a = self.inputs[0]
+        op = DataFrameIlocSetItem(indexes=self.indexes, value=new_output,
+                                  dtypes=a.dtypes, gpu=a.gpu, sparse=a.issparse(), object_type=self.object_type)
+        return op(a, self._indexes, new_output)
+
+    @classmethod
+    def tile(cls, op):
+        in_df = op.inputs[0]
+        out_val = op.outputs[0]
+
+        from ...tensor.indexing.getitem import TensorIndex, _getitem
+        from ...tensor.datasource.empty import empty
+
+        # See Note [Fancy Index of Numpy and Pandas]
+        tensor0 = _getitem(empty(in_df.shape[0], chunk_size=(in_df.nsplits[0],)).tiles(), op.indexes[0]).single_tiles()
+        tensor1 = _getitem(empty(in_df.shape[1], chunk_size=(in_df.nsplits[1],)).tiles(), op.indexes[1]).single_tiles()
+
+        out_chunks = []
+        for index_chunk, column_chunk in itertools.product(tensor0.chunks, tensor1.chunks):
+            # The indexes itself cannot be a Tensor of mars, that requires more complicate logic
+            if not isinstance(index_chunk.op, TensorIndex) or not isinstance(column_chunk.op, TensorIndex):
+                raise NotImplementedError('The index value cannot be unexecuted mars tensor')
+
+            in_chunk = in_df.cix[index_chunk.inputs[0].index + column_chunk.inputs[0].index]
+
+            chunk_op = op.copy().reset_key()
+            chunk_op._indexes = (index_chunk.op.indexes[0], column_chunk.op.indexes[0])
+
+            if isinstance(op.indexes[0], Integral) or isinstance(op.indexes[1], Integral):
+                # NB: pandas only compresses the result to series when index on one of axis is integral
+                if isinstance(op.indexes[1], Integral):
+                    shape = index_chunk.shape
+                    index = index_chunk.index
+                    dtype = in_chunk.dtypes.iloc[column_chunk.op.indexes[0]]
+                    index_value = indexing_index_value(in_chunk.index_value, index_chunk.op.indexes[0])
+                else:
+                    shape = column_chunk.shape
+                    index = column_chunk.index
+                    dtype = find_common_type(in_chunk.dtypes.iloc[column_chunk.op.indexes[0]])
```

Comment:
Should the series chunk's dtype be identical to the dataframe one?

Assume we have a df with three columns which dtypes is [int64, int64, float64], and chunk_size=2, the first series chunk's dtype would int64, and second be float64, but they should be all float64.

Reply:
The insight here is that, the dataframe is chunked, when for `[int64, int64, float64]`, when we do `.iloc` on the first chunk, we do get a series of type `int64`, rather than `flaot64`, so we mark the chunk's dtype as `int64`.

It won't affect the correctness of the final result and I just think this behavior is more accurate.

Reply:
I am worry about maybe some element-wise operation which is called after iloc may have wrong result. Because the actual dtyps is different for a chunk. Would this situation be possible in your opinion?

Reply:
I see. The `iloc` maybe fused with subsequent operation on the resulting series. I think here a `astype` is needed in the execution part.

Will fix it soon.